### DISCUSSION TITLE
Any idea how to carry over event subscriptions to the event manager for a new transaction?

### DISCUSSION THREAD

**Discussion by rat-matheson** [mikro-orm/mikro-orm#3215] [Q&A] [answered]
I would like to add events from the default EntityManager's event manager to the EntityManager created from the transaction method.  

```typescript
this.defaultEm.transactional((em) => { 
    //TODO: need to add all listeners from defaultEm somehow

    // Add a test to verify that event managers are different.  Want to account for a behavioural change
    if (em.getEventManager() == self.defaultEm.getEventManager()) {
        throw new Error("library changed...event manager now shared")
    }

    //want something like this
    self.defaultEm.getEventManager().subscribers.forEach(sub => em.getEventManager().registerSubscriber(sub));

    // continue with transaction
}
```

I was thinking maybe there is an option in transactional but I don't see it.

It looks like I could use ```(<Map<any,any>>this.defaultEm.getEventManager()["entities"]).keys()``` but then I'm accessing a private member which is not recommended

**Comment by B4nan** [comment, marked as answer]
I guess you are talking about some dynamically added filters, right? Because the ones from ORM config will be already there. And that is what you can leverage - when you add a new filter dynamically, add it also to the configuration object:

```ts
em.config.get('subscribers').push(...);
```

**Reply by rat-matheson** [reply]
Yes, I am talking about dynamically added EventSubscribers.  If I add a subscriber dynamically using ```em.config.get('subscribers').push(myDynamicEventSub)```, do I still need to also call ```em.getEventManager().registerSubscriber(this.myDynamicEventSub)```?

**Reply by rat-matheson** [reply]
Interesting observation.  If I update the config as you suggest, the dynamic subscribers are automatically added to new transactions.  Is this the intended behavior?  If so, then I don't have to add them to the transactional EntityManager which is nice.

```typescript
this.defaultEm.registerSubscriber(myDynamicEventSub);
this.defaultEm.config.get('subscribers').push(myDynamicEventSub);

//... and later on

this.defaultEm.transactional(em => {
    // I no longer need to do something like as follows because it would add duplicate subscribers, right?
    // this.defaultEm.config.get('subscribers').forEach(sub => em.getEventManager.registerSubscriber(sub))

    // do stuff
});
```

**Reply by B4nan** [reply]
Yes, correct, that's why I am suggesting this.

**Reply by rat-matheson** [reply]
Thanks.   Great to know this is possible!
